SoC/gd32vf103/Board/gd32vf103v_longan_nano/Include/gd32vf103v_longan_nano.h

```diff
@@ -0,0 +1,109 @@
+
+    1. Redistributions of source code must retain the above copyright notice, this
+       list of conditions and the following disclaimer.
+    2. Redistributions in binary form must reproduce the above copyright notice,
+       this list of conditions and the following disclaimer in the documentation
+       and/or other materials provided with the distribution.
+    3. Neither the name of the copyright holder nor the names of its contributors
+       may be used to endorse or promote products derived from this software without
+       specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+OF SUCH DAMAGE.
+*/
+
+#ifndef GD32VF103C_LONGAN_NANO_H
+#define GD32VF103C_LONGAN_NANO_H
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+#include "nuclei_sdk_soc.h"
+
+/* exported types */
+typedef enum
+{
+    LED1 = 0,
+    LED2 = 1,
+    LED3 = 2,
+}led_typedef_enum;
+
+typedef enum
+{
+    KEY_WAKEUP = 0,
+}key_typedef_enum;
+
+typedef enum
+{
+    KEY_MODE_GPIO = 0,
+    KEY_MODE_EXTI = 1
+}keymode_typedef_enum;
+
+
+/* sipeed longan nano board low layer led */
+#define LEDn                             (3U)
+
+#define LEDG_PIN                         GPIO_PIN_1
+#define LEDG_GPIO_PORT                   GPIOA
+#define LEDG_GPIO_CLK                    RCU_GPIOA
+
+#define LEDB_PIN                         GPIO_PIN_2
+#define LEDB_GPIO_PORT                   GPIOA
+#define LEDB_GPIO_CLK                    RCU_GPIOA
+
+#define LEDR_PIN                         GPIO_PIN_13
+#define LEDR_GPIO_PORT                   GPIOC
+#define LEDR_GPIO_CLK                    RCU_GPIOC
+
+/* sipeed longan nano board UART com port */
+#define GD32_COM0                        USART0
+#define GD32_COM_CLK                     RCU_USART0
+#define GD32_COM_TX_PIN                  GPIO_PIN_9
+#define GD32_COM_RX_PIN                  GPIO_PIN_10
+#define GD32_COM_TX_GPIO_PORT            GPIOA
+#define GD32_COM_RX_GPIO_PORT            GPIOA
+#define GD32_COM_TX_GPIO_CLK             RCU_GPIOA
+#define GD32_COM_RX_GPIO_CLK             RCU_GPIOA
+
+/* sipeed longan nano board low layer button (none)*/
+#define KEYn                             (0U)
+
+/* function declarations */
+/* configure led GPIO */
+void gd_longan_nano_led_init(led_typedef_enum lednum);
```

Comment:
This is a general question to @fanghuaqi and not directly about this port. Why do the function names have to be prefixed with `gd_board_name_somethin()`? Wouldn't it  make more sense to name this function `gd_led_init()` in all ports so that one could actually use the same code with different boards?

Reply:
I think this is a good suggestion, previously I was thinking the board level api might not be compatiable with each other, since the onboard resources maybe not be same.

Maybe we can think about generic APIs for gd32v related board level APIs.

Any proposal? maybe you can open an issue to discuss about it, and make a PR about it